core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -75,6 +75,7 @@
 
     private RetryPolicy retryPolicy = RetryPolicy.NONE;
     private int retryAttempts = 4;
+    private int retryAttemptsPerTask = Integer.MAX_VALUE;
```

Comment:
Make sense. 
It feels that I should also rename config property and session property to `query-retry-attempts` and `query_retry_attempts`.
Do you think we should also change `retry-initial-delay` and `retry-max-delay` to have `query-` prefix (and matching session properties)? Feels yeah to me but maybe you think otherwise.

Reply:
Yeah, it makes sense

Reply:
Well - actually no - the proposed naming suggests different meaning. We still limit number of task retries by `retryAttempts` just for any given task we allow up to `retryAttemptsPerTask`. 
Then I think the names I proposed are better (though kinda ugly) - still no idea for an improvement.

Reply:
I'm not sure if we really want to limit the total number of retries. It feels like the number of retries per task should be sufficient. Thus the `retryAttempts` is more of a `query-retry` specific, thus worth renaming. If we want to cap the total number of task retries do you think it makes sense to have a dedicated property? e.g.: `task-retry-max-attempts-overall`? `task-retry-max-attempts-per-task`?

Reply:
> e.g.: task-retry-max-attempts-overall? task-retry-max-attempts-per-task?

I like it. Will change